R/watcher.r

```diff
@@ -47,11 +47,10 @@ watch <- function(path, callback, pattern = NULL, hash = TRUE) {
 #' @return a digest of the file, or NA if it doesn't exist.
 #' @keywords internal
 safe_digest <- function(path) {
-  if (!file.exists(path)) return(NA_character_)
-  if (is_directory(path)) return(NA_character_)
-  if (!is_readable(path)) return(NA_character_)
-
-  digest::digest(path, file = TRUE)
+  tryCatch(
+    digest::digest(path, file = TRUE),
+    error = function(e) NA_character_
+  )
```

Comment:
I'll remove this patch for now. Still, short of trying, there seems to be no way to safely tell if `digest()` will succeed.